src/sdk/server.ts

```diff
@@ -92,32 +98,33 @@ export function createPlugin<TConfig = unknown, TEnv = unknown, TSupportedEvents
       env = ctx.env as TEnv;
     }
 
+    const workerUrl = new URL(inputs.ref).origin;
+    let workerName;
+
+    if (workerUrl.includes("localhost")) {
+      workerName = "localhost";
+    } else {
+      workerName = `${workerUrl.split("//")[1].split(".")[0]}`;
+    }
+
     const context: Context<TConfig, TEnv, TSupportedEvents> = {
       eventName: inputs.eventName as TSupportedEvents,
       payload: inputs.eventPayload,
       octokit: new customOctokit({ auth: inputs.authToken }),
       config: config,
       env: env,
       logger: new Logs(pluginOptions.logLevel),
+      pluginDeploymentDetails: `${workerName}`,
     };
 
     try {
       const result = await handler(context);
       return ctx.json({ stateId: inputs.stateId, output: result });
     } catch (error) {
       console.error(error);
-
-      let loggerError: LogReturn | null;
-      if (error instanceof Error) {
-        loggerError = context.logger.error(`Error: ${error}`, { error: error });
-      } else if (error instanceof LogReturn) {
-        loggerError = error;
-      } else {
-        loggerError = context.logger.error(`Error: ${error}`);
-      }
-
+      const loggerError = error as LogReturn | Error | null;
       if (pluginOptions.postCommentOnError && loggerError) {
-        await postComment(context, loggerError);
+        await postComment(context, loggerError, honoEnvironment);
```

Comment:
I've passed it as a param here instead of building it into `context` because the SDK consumer can access anything in `context` and if they installed their worker into any config other than their own they'd be able to exfiltrate the api token of the partner or us